### ISSUE TITLE
Renter tries to upload directory node

### ISSUE THREAD

**Issue by alexf91** [NebulousLabs/Sia#1855]
I have a directory with a few files in it:
```
foodir
├── bardir
│     └── zoofile
├── barfile1
├── barfile2
└── barfile3
```

If I upload this directory using `siac renter upload foodir foodir`, not only the files, but also the directory nodes are uploaded. The upload of the nodes never finishes.

```
siac renter list                
Tracking 6 files:
   4.1 KB  foodir (uploading, 0.00%)
   4.1 KB  foodir/bardir (uploading, 0.00%)
      0 B  foodir/bardir/zoofile
      0 B  foodir/barfile1
      0 B  foodir/barfile2
      0 B  foodir/barfile3
```

Related to #913 and #1334. Recursive upload works, but inodes are not ignored.

**Comment by lukechampine** [comment]
This should be a simple fix. I believe we just need to check for `info.IsDir` when walking the directory.

However, it also seems like the backend is not handling directories properly either. It should return an error if you attempt to upload one.

**Comment by johnathanhowell** [comment]
Fixed in #1856
